gnucash/gnome/dialog-trans-assoc.c

```diff
@@ -299,7 +305,10 @@ get_trans_info (AssocDialog *assoc_dialog)
     /* Get list of Accounts */
     accts = gnc_account_get_descendants_sorted (root);
 
+    /* disconnect the model from the treeview */
     model = gtk_tree_view_get_model (GTK_TREE_VIEW(assoc_dialog->view));
+    g_object_ref (G_OBJECT(model));
+    gtk_tree_view_set_model (GTK_TREE_VIEW(assoc_dialog->view), NULL);
```

Comment:
Same remark as with the find account dialog. You don't have to get-clear-set the model (and it seems to me you missed the clear step in this case). Just create a new model and set it.